### PR TITLE
Simple form demo

### LINKED ISSUE #689: Add action section to the form component
The spacing for the actions on the bottom of the form differ from the standard `pf-c-form__group` so a new section is required.

### PR DESCRIPTION
**Pull request by matthewcarleton**
This is a big PR that covers a lot of form stuff. :)

This covers the Form demo which I've named basic forms. I believe the plan is to break the complex forms out into a separate demo. If so, I believe that is where some further work may need to be considered for nested form groups. After talking with @andresgalante we concluded that it makes the most sense to not include sizing for groups at this time. 

I'm also not sure if using the toolbar layout is the best method for the action section of the form. The spacing is not the same as the form designs, but I wonder if we should align those better. Also, when using the toolbar there are spacing issues on smaller screens. Regardless of if this gets used in the form we should consider correcting this.

The trickiest part of these forms is the horizontal form. I'm sure there could be a better approach so open to feedback on that.

Also, @mceledonia I currently have a width of 150px on the labels, can you confirm what you'd like that to be. 

I've also updated a lot of the handlebars formatting where I could. 

I've expanded the check component to include a label and broke out the handlebars templates for that.

@jgiardino I've tested all of these with voiceover for now but would like to discuss a11y further with you and get a jaws run through on it.

For the forms component I removed all of the examples that were already included in their own component pages and just showed label alignment, help text and the action section modifier. 

Thanks for taking the time to review this :) 

This closes #689 #200

### CODE REVIEW

File: src/patternfly/demos/BasicForms/examples/alternative-form-demo-example.hbs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  {{#> form-group}}
+    {{#> form-label form-label--attribute='for="alt-form-git-repo"' required="true"}}Email{{/form-label}}
+    {{#> form-control pf-c-form-control--modifier="pf-m-form-control-alt" controlType="input" input="true" type="email" id="alt-form-git-repo" name="simple-form-git-repo" required="true"}}
+    {{/form-control}}
+  {{/form-group}}
+  {{#> form-group}}
+    {{#> form-label form-label--attribute='for="alt-form-password"' required="true"}}Password{{/form-label}}
+    {{#> form-control pf-c-form-control--modifier="pf-m-form-control-alt" controlType="input" input="true" type="password" id="alt-form-password" name="alt-form-password" aria-describedby="alt-helper3" required="true"}}
+    {{/form-control}}
+    {{#> form-helper-text form-helper-text--attribute='id="alt-helper3" aria-live="polite"'}}Must be a minium of 6 alhpa-numeric characters.{{/form-helper-text}}
```
Comment (andybraren) [inline src/patternfly/demos/BasicForms/examples/alternative-form-demo-example.hbs]: Spelling: "Must be a minimum of 6 alpha-numeric characters."

**Comment by mcarrano** [comment]
@matthewcarleton I just reviewed this and definitely think that it does a good job illustrating most of the considerations for forms in PatternFly. Your changes make sense, although wasn't sure why Actions are part of a Form component and not just something that can be used anywhere, e.g. as part of a data list or a toolbar. Just a few additional thoughts/comments:

* For the horizontal form, I was wondering what will happen if I compress this to mobile. Does not seem to be responsive right now.
* It would make sense to also consider the placement if info tips (i.e. field level help) as part of this demo. That would utilize a popover component. We should discuss where this lives.
* I wasn't sure where field level validation should be demoed. Looks like there are examples of this in the component called Form. We still need to sort out the difference between the Form component and the Form demo. Not sure this will be clear to visitors to the PatternFly site.

**Comment by matthewcarleton** [comment]
hey @mcarrano thanks for the review.

So actions are just for forms at the moment because it is just adding a top spacer to the toolbar that is used at the bottom of the form. This has different spacing from the standard form group.

The horizontal form does stack but you'll have to open in a new window and resize your browser to really see it. I can adjust the breakpoint for this, we also need to confirm what the label width is.

I think tooltip, along with any kind of action needs to be demo'd in the React workspace. We can have a static example of things but I'm not sure how useful that is. Happy to discuss this more. This would include field level help as well. We need to introduce JS to really demo that.

I didn't do it, But I don't mind the idea of the form component being called form(layout) or something that explains it from that high level. Within the form component it might be useful to also link to each component as well as the demos that we put together. That would help cover things. I see the form component as a place where we show layout, labels, messages and docs.

Hope all that makes sense.

**Comment by jgiardino** [comment]
Hey @mcarrano, 
@matthewcarleton and I met last week to review what form demos we should create to support testing for accessibility. We created an outline of demos in [this document](https://docs.google.com/document/d/1YNLjxGTCNhKe4VSGxEZcuwCkFNIayFjwx-IV7HlYzEo/edit?usp=sharing). Demo #4 includes some of the items you're asking about, like field level validation and infotips.

That said, I'm not sure if the demo we create for testing accessibility will persist when we're done testing for accessibility. And I'm not opposed to updating the outline we have to better align with the demos we want to provide.

**Comment by mcarrano** [comment]
@matthewcarleton @jgiardino Thank you both for the feedback. The good news is that I think we are converging on similar things. Here is a link to a document that I was working on that outlines what I think this needs to look like from a PF website and design guidance perspective: https://docs.google.com/document/d/1kzUsco3USvzbi-Qhlzat13Coil7s6Lmrqwi6CcvlMCk/edit?usp=sharing  @jgiardino You will find many of the same topics and concerns for a demo as exist in your document.

@matthewcarleton I do get what you are saying about some of these items needing JS to demo. Not sure how we address this as it will likely come up elsewhere. Maybe for help/info tips we add in the icon, but it doesn't activate a pop-up except in the React demo.

I will try to coordinate a time that we can all talk further on how to structure.

**Comment by jgiardino** [comment]
@matthewcarleton These look good! I reviewed these demos in JAWS and noted what gets announced in the [forms a11y review doc](https://docs.google.com/document/d/1YNLjxGTCNhKe4VSGxEZcuwCkFNIayFjwx-IV7HlYzEo/edit?usp=sharing). 

The only issue I noticed is with using a number input type for a phone number. When I entered a number like 919-555-5555, JAWS told me that value was invalid.

These are a couple of things I'd like to test in follow-up demos just to have a better understanding:

- There was some odd behavior with the text area in the second form element. I don't know if this behavior is a result of the select element, or the text area. I don't see anything odd in the html, and I think this is just how JAWS automatically switches in and out of [forms mode](https://www.freedomscientific.com/Training/Surfs-Up/Forms.htm).

- So far, my only experience with `aria-describedby` is that the text isn't announced automatically but JAWS will let me know that it's available. But in this case, it is announced automatically in addition to JAWS letting me know that it's available. And I wonder if that's due to the presence of `aria-live`. I'd like to test whether removing `aria-live` from the field description changes what gets announced. The [webAIM example](https://webaim.org/techniques/forms/advanced#describedby) for description text doesn't include this. Do you remember where you found this example.

**Comment by matthewcarleton** [comment]
Thanks @jgiardino I've updated the simple demo to use text="tel" instead of number which is what I should have used to begin with. The nice thing about tel is you get a number keyboard on mobile .

I've also added placeholder text here as a way to indicate format. Curious how JAWS interprets that. I don't think using placeholder on every field is necessary, it would likely be frustrating for a screen reader user but in this case it could be useful.

File: src/patternfly/components/Check/check-input.hbs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<input class="pf-c-check__input{{#if check-input--modifier}}{{check-input--modifier}}{{/if}}"
```
Comment (michael-coker) [inline src/patternfly/components/Check/check-input.hbs]: You're missing a space, should be `{{#if check-input--modifier}} {{check-input--modifier}}`

File: src/patternfly/components/Check/check-label.hbs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<label class="pf-c-check__label{{#if check-label--modifier}} {{check-label--modifier}}{{/if}}"
+  {{#if check-label--attribute}}
+  {{{check-label--attribute}}}
+{{/if}}
+>
```
Comment (michael-coker) [inline src/patternfly/components/Check/check-label.hbs]: Can you move this up a line, after `{{/if}}` to be consistent?

File: src/patternfly/components/Check/check.hbs [hunk 4]
```diff
@@ -1,13 +1,4 @@
...
-  {{#if checked}} checked{{/if}}
-  {{#if tabindex}} tabindex="{{tabindex}}" {{/if}}
-  {{#if aria-describedby}} aria-describedby="{{aria-describedby}}" {{/if}}
-  {{#if aria-invalid}} aria-invalid="true" {{/if}}
-  {{#if aria-label}} aria-label="{{aria-label}}" {{/if}}
-  {{#if aria-errormessage}} aria-errormessage="{{aria-errormessage}}"{{/if}}
-  {{#if required}} required aria-required="true" {{/if}}
-  >
\ No newline at end of file
+<div class="pf-c-check{{#if check-modifier}}{{check--modifier}}{{/if}}">
```
Comment (michael-coker) [inline src/patternfly/components/Check/check.hbs]: missing a space, should be `{{#if check-modifier}} {{check--modifier}}`

File: src/patternfly/components/Check/styles.scss [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+  --pf-c-check__label--MarginLeft: var(--pf-global--spacer--sm);
+  --pf-c-check__label--diabled--color: var(--pf-global--disabled-color--100);
 
-  &:disabled {
+  .pf-c-check__label.pf-m-disabled, .pf-c-check__input:disabled {
+    color: var(--pf-c-check__label--diabled--color);
     cursor: not-allowed;
   }
+
+  .pf-c-check__input, .pf-c-check__label {
```
Comment (michael-coker) [inline src/patternfly/components/Check/styles.scss]: can you put `.pf-c-check__label` on it's own line? I'll look to see if I can create a stylelint rule for this.

File: src/patternfly/components/Form/examples/form-action-group-example.hbs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{{#> form}}
+  {{#> form-group form-group--modifier="pf-m-action"}}
+        {{#> toolbar}}
```
Comment (michael-coker) [inline src/patternfly/components/Form/examples/form-action-group-example.hbs]: you have an extra tab indent for this block

File: src/patternfly/components/Form/form-action-group.hbs [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<div class="pf-c-form__action-group{{#if form-action-group--modifier }} {{ form-action-group--modifier }}{{/if}}">
```
Comment (michael-coker) [inline src/patternfly/components/Form/form-action-group.hbs]: remove the spaces in `{{ form-action-group--modifier }}`

Comment (michael-coker) [reply src/patternfly/components/Form/form-action-group.hbs]: looks like there are a lot of instances of `{{ ` and ` }}` in this PR. It's just a syntax thing and I don't know that we have a standard for it. fwiw, there are no spaces in the [contribution guide](https://deploy-preview-728--pf-next.netlify.com/contribution). I'll leave it to you if you want to remove them or not :)

File: src/patternfly/components/Form/form-fieldset.hbs [hunk 8]
```diff
@@ -1,3 +1,3 @@
-<fieldset class="pf-c-form__fieldset{{#if pf-c-form__fieldset--modifier}} {{ pf-c-form__fieldset--modifier }}{{/if}}">
+<fieldset class="pf-c-form__fieldset{{#if form__fieldset--modifier}} {{ form__fieldset--modifier }}{{/if}}">
```
Comment (michael-coker) [inline src/patternfly/components/Form/form-fieldset.hbs]: this should be `form-fieldset--modifier`

File: src/patternfly/components/Form/form-helper-text.hbs [hunk 9]
```diff
@@ -1,7 +1,7 @@
-<p class="pf-c-form__helper-text{{#if pf-c-form__helper--modifier}} {{ pf-c-form__helper--modifier }}{{/if}}" 
-{{#if id}} id="{{id}}"{{/if}}
-{{#if aria-live}} aria-live="{{aria-live}}"{{/if}}
-{{#if aria-hidden}} aria-hidden="{{aria-hidden}}" {{/if}}
+<p class="pf-c-form__helper-text{{#if form-helper-text--modifier}} {{ form-helper-text--modifier }}{{/if}}" 
+{{#if form-helper-text--attribute}}
+{{{ form-helper-text--attribute }}}
+{{/if}}
 >
```
Comment (michael-coker) [inline src/patternfly/components/Form/form-helper-text.hbs]: can you move this up a line after `{{/if}}`?

File: src/patternfly/components/Form/form-label.hbs [hunk 10]
```diff
@@ -1,5 +1,10 @@
-<label class="pf-c-form__label{{#if pf-c-form__label--modifier}} {{ pf-c-form__label--modifier }}{{/if}}{{#if required}} pf-m-required{{/if}}{{#if hidden}} pf-u-sr-only{{/if}}" 
-{{#if for}} for="{{for}}"{{/if}} 
+<label class="pf-c-form__label{{#if form-label--modifier}} {{ form-label--modifier }}{{/if}}{{#if hidden}} pf-u-sr-only{{/if}}" 
+{{#if form-label--attribute}}
+{{{form-label--attribute}}}
```
Comment (michael-coker) [inline src/patternfly/components/Form/form-label.hbs]: needs an indent

File: src/patternfly/components/Form/form-label.hbs [hunk 11]
```diff
@@ -1,5 +1,10 @@
...
+{{/if}} 
 >
```
Comment (michael-coker) [inline src/patternfly/components/Form/form-label.hbs]: Move this up after `{{/if}}` please

File: src/patternfly/demos/BasicForms/examples/horizontal-form-demo-example.hbs [hunk 12]
```diff
@@ -0,0 +1,83 @@
+{{#> horizontal-form-demo}}
```
Comment (michael-coker) [inline src/patternfly/demos/BasicForms/examples/horizontal-form-demo-example.hbs]: The indentation in this file is off.

File: src/patternfly/demos/BasicForms/examples/simple-form-demo-example.hbs [hunk 13]
```diff
@@ -0,0 +1,59 @@
+{{#> simple-form-demo}}
```
Comment (michael-coker) [inline src/patternfly/demos/BasicForms/examples/simple-form-demo-example.hbs]: indentation in this file is off.

File: src/patternfly/demos/BasicForms/examples/alternative-form-demo-example.hbs [hunk 14]
```diff
@@ -0,0 +1,41 @@
+{{#> alternative-form-demo}}
```
Comment (michael-coker) [inline src/patternfly/demos/BasicForms/examples/alternative-form-demo-example.hbs]: indentation in this file is off.

**Comment by michael-coker** [comment]
in the `code/docs.md` file, can you give the form demo a description and remove the accessibility/usage sections?

File: src/patternfly/components/Form/examples/form-action-group-example.hbs [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+        {{#> toolbar-group}}
+          {{#> toolbar-item}}
+            {{#> button btnClass="pf-m-secondary"}}
+              Cancel
+            {{/button}}
+          {{/toolbar-item}}
+        {{/toolbar-group}}
+      {{/toolbar}}
+  {{/form-group}}
+{{/form}}
```
[the remark this answers is not in the document]
Comment (matthewcarleton) [reply src/patternfly/components/Form/examples/form-action-group-example.hbs]: hey, yes I notice this as well and was planning on creating an issue for it. I did discuss a bit with others. I can open the issue if you like or feel free to do it :)

File: src/patternfly/components/Form/styles.scss [hunk 16]
```diff
@@ -1,217 +1,163 @@
...
+  --pf-c-form__label__requried--MarginLeft: var(--pf-global--spacer--xs);
+  --pf-c-form__label__requried--FontSize: var(--pf-global--FontSize--sm);
+  --pf-c-form__label__requried--Color: var(--pf-global--danger-color--100);
 
-  // Legend
-  --pf-c-form__legend--MarginBottom: var(--pf-global--spacer--md);
-  --pf-c-form__legend--FontSize: var(--pf-global--FontSize--lg);
-  --pf-c-form__legend--FontWeight: var(--pf-global--FontWeight--semi-bold);
+  // action group
+  --pf-c-form--action-group--MarginTop: var(--pf-global--spacer--xl);
```
[the remark this answers is not in the document]
Comment (matthewcarleton) [reply src/patternfly/components/Form/styles.scss]: do you mean --pf-c-form__group--m-action--MarginTop?

[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Form/styles.scss]: Oof, yessir that's what I mean

File: src/patternfly/components/Form/examples/form-action-group-example.hbs (hunk 15 above)
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Form/examples/form-action-group-example.hbs]: Awesome, thanks!

**Comment by jgiardino** [comment]
I would still keep `type="tel"`. Even though there aren't any benefits that I can notice in JAWS, it also doesn't make the experience worse. And there are still other benefits for other users, like on mobile devices as you mention earlier. 

The placeholder is the one that makes me a little dyspeptic*. I was chatting with @mceledonia about the styles for that (which is outside the scope of this PR), but while looking for recommendations, I am noticing that placeholder text is prefixed with a "E.g." or "For example,". Here are some of these examples: 
- https://inclusive-components.design/a-todo-list/#labeling
- https://www.w3.org/WAI/GL/wiki/Using_@placeholder_on_input

In the context of a simple form like this one (i.e. standard label/input layout), does it make sense to have `placeholder="For example, 555-555-5555"`? Maybe that helps to solve the challenge of how to differentiate placeholder text from inputs, both visually and when using a screen reader. 

*It's talk like a pirate day, and my [pirate name](http://gangstaname.com/names/pirate/generate) is Dispeptic Helga Bonny

**Comment by mattnolting** [comment]
Beautiful!

**Pull request merged**